src/main.rs

```diff
@@ -12,22 +14,47 @@ enum Command {
     Login(login::ContestInfo),
 
     #[structopt(name = "submit", about = "Submits FILE to given PROBLEM.")]
-    Submit(submit::Submission),
+    Submit(client::Submission),
+}
+
+// TODO: Update all necessary code to support any judge system, not only Ejudge one
+type AnyClient = ejudge::EjudgeClient;
```

Comment:
Maybe, we need base trait with `submit` only for `Client` and Box object implementing it?